zvariant/src/lib.rs

```diff
@@ -1177,7 +1177,7 @@ mod tests {
         let v: Value<'_> = dict.into();
         assert_eq!(v.value_signature(), "a{sv}");
         let encoded = to_bytes(ctxt, &v).unwrap();
-        assert_eq!(dbg!(encoded.len()), 68);
+        assert_eq!(dbg!(encoded.len()), 66);
```

Comment:
In the original code where the order of the `Dict` entries mattered, the internal entries looked like this:
```
       [ { key: "hello", value: "there"}, { key:"bye", "now"} ]
```
with the change to `BTreeSet` they are reordered to
```
       [  { key:"bye", "now"}, { key: "hello", value: "there"} ]
```

The serializer (zvariant/src/dbus/ser.rs:134) is adding various paddings for alignment. I am quite certain that this is the reason the len of the encoded bytes goes down but the content remains correct (see following lines of the test).